flow/hooks/track_operations.py

```diff
@@ -0,0 +1,164 @@
+# Copyright (c) 2023 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+"""Built in execution hook for basic tracking."""
+import json
+
+from .util import collect_metadata
+
+try:
+    import git
+except ImportError:
+    GIT = False
+else:
+    from .git_util import collect_metadata_with_git
+
+    GIT = True
+
+
+FN_LOGFILE = ".operations_log.txt"
+
+
+class TrackOperations:
+    """:class:`~.TrackOperations` tracks information about the execution of operations to a logfile.
+
+    This hooks can provides information on the start, successful completion, and/or
+    erroring of one or more operations in a `flow.FlowProject` instance. The logs are stored in a
+    file given by the parameter ``fn_logfile``. This file will be appended to if it already exists.
+    The default formating for the log provides the [time, job id, log level, and log message].
+    .. note::
+        All tracking is performed at the INFO level. To ensure outputs are captured in log files,
+        use the `--debug` flag when running or submitting jobs, or specify
+        `submit_options=--debug` in your directives (example shown below).
+
+    Examples
+    --------
+    The following example will install :class:`~.TrackOperations` at the operation level.
+    Where the log will be stored in a file name `foo.log` in the job workspace.
+
+    .. code-block:: python
+        from flow import FlowProject
+        from flow.hooks import TrackOperations
+
+
+        class Project(FlowProject):
+            pass
+
+
+        def install_operation_track_hook(operation_name, project_cls):
+            track = TrackOperation(f"{operation_name}.log")
+            return lambda op: track.install_operation_hooks(op, project_cls)
+
+
+        @install_operation_log_hook("foo", Project)
+        @Project.operation(directives={
+            "submit_options": "--debug"  # Always submit operation foo with the --debug flag
+        })
+        def foo(job):
+            pass
+
+
+    The code block below provides an example of how install :class:`~.TrackOperations` to a
+    instance of :class:`~.FlowProject`
+
+    .. code-block:: python
+        from flow import FlowProject
+        from flow.hooks import TrackOperations  # Import build
+
+
+        class Project(FlowProject):
+            pass
+
+
+        if __name__ == "__main__":
+            project = Project()
+            project = TrackOperations().install_project_hooks(project)
+            project.main()
+
+
+    Parameters
+    ----------
+    fn_logfile: str
```

Comment:
I don't see this parameter in the constructor (`__init__` below). Also the `strict_git` argument is not documented.